oeps/oep-0039-yaml-config.rst

```diff
@@ -0,0 +1,146 @@
++-----------------+--------------------------------------------------------+
+| Authors         | Cory Lee - cory@edx.org                                |
++-----------------+--------------------------------------------------------+
+| Arbiter         | TBA                                                    |
++-----------------+--------------------------------------------------------+
+| Status          | Draft                                                  |
++-----------------+--------------------------------------------------------+
+| Type            | Best Practice                                          |
++-----------------+--------------------------------------------------------+
+| Created         | 2019-07-15                                             |
++-----------------+--------------------------------------------------------+
+| `Review Period` | <start - target end dates for review>                  |
++-----------------+--------------------------------------------------------+
+| `Resolution`    |                                                        |
++-----------------+--------------------------------------------------------+
+| `References`    |                                                        |
++-----------------+--------------------------------------------------------+
+
+Abstract
+========
+
+Yaml configuration instead of python django configuration settings files for configuring django applications is preferable because it simplifies configuration of django applications
+
+Motivation
+==========
+
+Currently there is no standard for this and many applications have varying takes on configuration. 
+Since Open edX consists of many applications managing and configuring it can be difficult, It would be much easier if all of the applications were configured consistently. 
+This OEP intends to provide a set of standard best practices for configuring Open edX in a python django application.
+
+Specification
+=============
+
+Your configuration should consist of two files
+
+settings.py
+defaults.py 
+
+defaults.py should list every configuration key in the application, and provide a description of what it does.
+
+settings.py should do two things:
+1) It should first load defaults.py into the python settings module.
+2) Then should load a single yaml file into the default keyspace overwriting values in the settings module.
+
+The yaml file path loaded should be configurable via environment variable that is specific to the application
+eg: ECOMMERCE_CFG. 
+
+There should be no additional django configuration settings files like devstack.py, testing.py, docker.py, docker_devstack.py, or anything of the sort.
+
+Plugins should not modify the loaded settings python module. Since we load the keys from the yaml file, even the ones we do not know at the application layer.
+A plugin should not need to modify the configuration of the application.
+
+All of the configuration should be done in the yaml file
```

Comment:
There are some types of django configuration that are not just lists and dicts (and yaml representable data). I think that that's probably fine, but it would be worth mentioning that anything that is not lists-and-dicts should be computed from the lists-and-dicts after it that is loaded from the yaml file.

Reply:
Yeah interpreting the config to create django objects to configure libraries and whatnot is totally reasonable. I will clarify.